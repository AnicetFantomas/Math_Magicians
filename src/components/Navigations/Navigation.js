import React from 'react';
import { NavLink } from 'react-router-dom';
import './navigation.css';

const Navigation = () => (
  <div className="Navigation-bar">
    <h1>Math Magician</h1>
    <div className="navLinks">
      <NavLink exact to="/" activeClassName="nav-active" className="link-el">Home</NavLink>
      <NavLink exact to="/Calculator" activeClassName="nav-active" className="link-el">Calculator</NavLink>
      <NavLink exact to="/Quotes" activeClassName="nav-active" className="link-el">Quotes</NavLink>
    </div>
  </div>
);

export default Navigation;
