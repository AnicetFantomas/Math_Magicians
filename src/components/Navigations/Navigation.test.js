import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from './Navigation';

describe('Home page of App', () => {
  test('Build the snapshot of the Home component safely', () => {
    const nav = render(
      <Router>
        <Navigation />
      </Router>,
    );
    expect(nav).toMatchSnapshot();
  });
});
