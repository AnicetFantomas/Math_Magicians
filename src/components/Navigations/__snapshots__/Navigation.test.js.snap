// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page of App Build the snapshot of the Home component safely 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Navigation-bar"
      >
        <h1>
          Math Magician
        </h1>
        <div
          class="navLinks"
        >
          <a
            activeclassname="nav-active"
            aria-current="page"
            class="link-el active"
            href="/"
          >
            Home
          </a>
          <a
            activeclassname="nav-active"
            class="link-el"
            href="/Calculator"
          >
            Calculator
          </a>
          <a
            activeclassname="nav-active"
            class="link-el"
            href="/Quotes"
          >
            Quotes
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Navigation-bar"
    >
      <h1>
        Math Magician
      </h1>
      <div
        class="navLinks"
      >
        <a
          activeclassname="nav-active"
          aria-current="page"
          class="link-el active"
          href="/"
        >
          Home
        </a>
        <a
          activeclassname="nav-active"
          class="link-el"
          href="/Calculator"
        >
          Calculator
        </a>
        <a
          activeclassname="nav-active"
          class="link-el"
          href="/Quotes"
        >
          Quotes
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
